src/terraform-aws/content.md

```diff
@@ -0,0 +1,713 @@
+
+Consequently, kubeadm provides the lowest degree of automation but the highest degree of flexibility.
+
+kubeadm is an executable that is run directly on the infrastructure on which the cluster is to be installed.
+
+In contrast to the other presented installation tools, kubeadm does not make any strong assumptions about the purpose of the cluster, such that it should be production-ready, or highly available.
+
+In particular, kubeadm performs the following tasks:
+
+- Creating and distributing client and server certificates for the individual Kubernetes components (API server, etcd, scheduler, etc.)
+- Creating and distributing kubeconfig files for the individual Kubernetes components (so that they can talk to the API server)
+- Launching the kubelet as a [systemd](https://systemd.io/) process
+- Launching the remaining Kubernetes components as Pods in the [host network](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2938) (managed by the kubelet)
+
+The result is a "minimum viable" cluster that you can then freely customise yourself according to your requirements.
+
+Moreover, kubeadm allows fine control over the individual cluster settings — for example, you can directly define command-line arguments for the individual Kubernetes components with a [kubeadm configuration file](https://pkg.go.dev/k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1beta2).
+
+_In short, kubeadm is an optimal solution if you want maximum control over the cluster creation process._
+
+However, the flip side is that creating a cluster with kubeadm is by no means done with a single command (as it's the case for kops and kubespray).
+
+Rather, it requires a whole series of manual steps:
+
+- Provision the infrastructure on which you want to create the cluster
+- Log in to every node and install kubeadm and Docker on it
+- Run the [`kubeadm init`](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/) command on one of the nodes
+- Run the [`kubeadm join`](https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-join/) command on all the other nodes
+
+These steps furthermore depend on each other — for example, the `kubeadm join` commands must include a token and other identifiers that are generated by the initial `kubeadm init` command.
+
+In summary, creating a Kubernetes cluster with kubeadm is a time-consuming tedious process when done manually, which counteracts the goal of having quick disposable clusters for running experiments.
+
+_This is where the Terraform kubeadm module comes in._
+
+## Terraform kubeadm module
+
+Learnk8s created the [Terraform kubeadm module](https://registry.terraform.io/modules/weibeld/kubeadm/aws) to combine the simplicity and flexibility of kubeadm with a degree of automation comparable to other higher-level tools.
```

Comment:
This sounds like a sales pitch.

I think you should say:

1. you can create your own kops
1. it is based on kubeadm
1. why not using terraform to build it?
1. hey, we've done it so you can copy from us.

The next of the article explains how, but here's a link to the working repo.